### PR TITLE
Improve log message in AssertIsValidForPubSub

### PR DESCRIPTION
**Pull request by danielmarbach**
From @zlepper as feedback in LiveChat.

This is a low hanging fruit so I PRed it directly.

We could also consider improving the exception messages in this class but I figured it would be a different PR to not distract from the scope of this PR

### CODE REVIEW

File: src/NServiceBus.Core/Routing/MessagingBestPractices/Validations.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public void AssertIsValidForPubSub(Type messageType)
 
             if (!conventions.IsEventType(messageType))
             {
-                Log.Info("You are using a basic message to do pub/sub, consider implementing the more specific ICommand and IEvent interfaces to help NServiceBus to enforce messaging best practices for you.");
+                Log.Info($"You are using a basic message to do pub/sub, consider implementing the more specific ICommand and IEvent interfaces on the type '{messageType}' to help NServiceBus to enforce messaging best practices for you.");
```
Comment (timbussmann) [inline src/NServiceBus.Core/Routing/MessagingBestPractices/Validations.cs]: should we maybe just display `messageType.FullName` though?

Comment (danielmarbach) [reply src/NServiceBus.Core/Routing/MessagingBestPractices/Validations.cs]: Done

**Comment by danielmarbach** [comment]
As mentioned https://github.com/Particular/NServiceBus/pull/5403

File: src/NServiceBus.Core/Routing/MessagingBestPractices/Validations.cs [hunk 2]
```diff
@@ -45,7 +45,7 @@ public void AssertIsValidForPubSub(Type messageType)
 
             if (!conventions.IsEventType(messageType))
             {
-                Log.Info("You are using a basic message to do pub/sub, consider implementing the more specific ICommand and IEvent interfaces to help NServiceBus to enforce messaging best practices for you.");
+                Log.Info($"You are using a basic message to do pub/sub, consider implementing the more specific ICommand and IEvent interfaces on the type '{messageType.FullName}' to help NServiceBus to enforce messaging best practices for you.");
```
Comment (bording) [inline src/NServiceBus.Core/Routing/MessagingBestPractices/Validations.cs]:
```suggestion
                Log.Info($"You are using a basic message to do pub/sub. Consider implementing the more specific ICommand and IEvent interfaces on the type '{messageType.FullName}' to help NServiceBus to enforce messaging best practices for you.");
```

**Pull request merged**